### ISSUE TITLE
Training Discriminator vs Generator in TUNIT

### ISSUE THREAD

**Issue by ammar-deep** [clovaai/tunit#30]
Could you please explain that why the gradients of `d_adv_real, d_gp, and d_adv_fake` are calculated separately when training the Discriminator? Whereas in G only the summation of all loses i.e. `g_loss` is used for calculating the gradients? 

I am referring to the following section of code.
https://github.com/clovaai/tunit/blob/49e633179c5670228a2319d2c99054322616b516/train/train_unsupervised.py#L139-L145

**Comment by FriedRonaldo** [comment]
In summary, it is just for the ease of debugging.

When I use distributed setting (ddp), if the output of the discriminator has multiple items, the ddp raises some issues about the gradient graph. To make the debug easy, I just split the loss functions into three parts. However, the separate update does not affect the performance.

**Comment by ammar-deep** [comment]
Alright Thanks.
